tests/backup/backup_psa_test.go

```diff
@@ -943,3 +944,460 @@ var _ = Describe("{RestoreFromHigherPrivilegedNamespaceToLower}", Label(TestCase
+				} else if strings.Contains(namespace, "baseline") {
+					namespaceMapBaselineToPrivilege[bkpNamespaces[i]] = namespace
+					err := CreateNamespaceAndAssignLabels(namespace, PrivilegedPSALabel)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %s", namespace))
+					log.Infof("Baseline to Privilege PSA map %v", namespaceMapBaselineToPrivilege)
+				}
+				restoredNamespaceList = append(restoredNamespaceList, namespace)
+				log.InfoD("Created namespace list on destination cluster %v", restoredNamespaceList)
+				projectName := fmt.Sprintf("project-%v-%d", RandomString(5), i)
+				_, err = Inst().S.(*rke.Rancher).CreateRancherProject(projectName, RancherProjectDescription, "vpinisetti-62", projectLabel, projectAnnotation)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating rancher project %s", projectName))
+				destClusterProjectList = append(destClusterProjectList, projectName)
+				log.Infof("The destination cluster project list is %v", destClusterProjectList)
+				projectID, err := Inst().S.(*rke.Rancher).GetProjectID(projectName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Getting Project ID - %s for project %s", projectID, projectName))
+				restoreProjectUIDMapping[projectID] = sourceNamespaceProjectUIDMapping[bkpNamespaces[i]]
+				log.Infof("The restore project UID mapping is %v", restoreProjectUIDMapping)
+				destClusterProjectUIDList = append(destClusterProjectUIDList, projectID)
+				log.Infof("The source cluster project UID list is %v", destClusterProjectUIDList)
+				restoreProjectNameMapping[projectName] = sourceNamespaceProjectMapping[bkpNamespaces[i]]
+				err = Inst().S.(*rke.Rancher).AddNamespacesToProject(projectName, []string{namespace})
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Adding namespace %s to project %s", namespace, projectName))
+				log.Infof("Project name mapping %v which is required to restore the backup is %v", restoreProjectNameMapping, bkpNamespaces[i])
+				err = Inst().S.(*rke.Rancher).ValidateProjectOfNamespaces(projectName, []string{namespace})
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying project %s of namespace %s", projectName, namespace))
+				destNamespaceProjectMapping[namespace] = projectName
+				log.InfoD("The destination namespace project mapping is %v", destNamespaceProjectMapping)
+			}
+		})
+
+		Step("Restore restricted to baseline namespaces with namespace & project mappings", func() {
+			log.InfoD("Restore restricted to baseline namespaces with namespace & project mappings")
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster failed")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.Infof("The restricted backup which is going to restore is %v", restrictedManualBackupName)
+			restoreName := fmt.Sprintf("restore-rtob-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
```

Comment:
this can be removed as restore are removed when cluster is deleted

Reply:
Done